tests/testthat/test-ratio_calculations.R

```diff
@@ -244,4 +244,52 @@ describe("calculate_ratios", {
     expect_equal(ratios_df$PPSTRES, rep(c(2 / 3), 4))
     expect_true(all(grepl("RACMAX", ratios_df$PPTESTCD)))
   })
+
+  it("handles early exit with no matches (data.frame) and returns correct structure", {
```

Comment:
nitpick: I guess now we don't do early exit now (?) depends on what you decided before
```suggestion
  it("handles no test and reference matches (data.frame) and returns correct structure", {
```